src/codemagic/firebase/api_client.py

```diff
@@ -0,0 +1,54 @@
+import json
+from pathlib import Path
+from typing import AnyStr
+from typing import List
+from typing import Optional
+from typing import Union
+
+from googleapiclient import discovery
+from oauth2client.service_account import ServiceAccountCredentials
+
+from .api_error import FirebaseAPIClientError
+from .resource_managers.release_manager import FirebaseReleaseResourceManager
+from .resources import FirebaseReleaseResource
+
+CredentialsJson = Union[AnyStr, dict, Path]
+
+
+class FirebaseAPIClient:
+    def __init__(self, service_account_json_keyfile: CredentialsJson):
+        self._service_account_json_keyfile = service_account_json_keyfile
+
+    @property
+    def service_account_json(self):
+        if isinstance(self._service_account_json_keyfile, dict):
+            return self._service_account_json_keyfile
+
+        if isinstance(self._service_account_json_keyfile, Path):
+            content = self._service_account_json_keyfile.read_text()
+        else:
+            content = self._service_account_json_keyfile
+
+        try:
+            return json.loads(content)
+        except ValueError as ve:
+            message = 'Unable to parse service account credentials, must be a valid json'
+            raise FirebaseAPIClientError(message) from ve
+
+    @property
+    def service(self):
+        credentials = ServiceAccountCredentials.from_json_keyfile_dict(self.service_account_json)
+        return discovery.build('firebaseappdistribution', 'v1', credentials=credentials)
+
+    def list_releases(
```

Comment:
To be more in line with ASC implementation you could use follow a pattern like
```python
@property
def releases(self) -> FirebaseReleaseResourceManager:
    ...
```
And then use it like
```python
client.releases.list(...)
```